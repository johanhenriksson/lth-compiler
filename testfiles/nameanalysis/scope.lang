int glob = 5;

function a(int one, float one) {
    int one;
    int two;
    three = 5;
    two = glob;
}

function b() {
    int one; // ok
    one = two; // error
    glob = one;
}
