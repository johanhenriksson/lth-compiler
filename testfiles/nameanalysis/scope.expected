Error at line 3,27: symbol 'one' is already declared!
Error at line 4,9: symbol 'one' is already declared!
Error at line 6,5: symbol 'three' is not declared
Error at line 12,11: symbol 'two' is not declared
